StRoot/StBTofUtil/StBTofSimResParams.h

```diff
@@ -20,80 +21,99 @@
+		 * then returns a single vertex resolution (in ps) for use in embedding w/ vpdStart
+		 */
+		double timeres_tof(uint itray, uint imodule, uint icell) 
+		{
+			double result = 8.5e-11;
+			if ( itray > 120 || imodule > 32 || icell > 6 )
+				return result;
 
-		return params[ itray ][ imodule * 6 + icell ];
-
-	}
-
-	//! Loads Vpd Sim Params from database
-
-	void loadParams(const int date = 20160913, const int time = 175725, const char* Default_time = "2016-09-13 17:57:25")
-	{
-
-		St_db_Maker *dbMk = dynamic_cast<St_db_Maker*>( GetChain()->GetMakerInheritsFrom("St_db_Maker") );
+			return params[ itray ][ imodule * 6 + icell ];
+		}
 
-		TDataSet *DB = GetDataBase("Calibrations/tof/tofSimResParams");
+		//! Loads BTOF Sim Params from database
+		//void loadParams(const int date = 20160913, const int time = 175725, const char* Default_time = "2016-09-13 17:57:25")
+		void loadParams(const int runNumber = 20076002)
+		{
+			
+			const int yearNumber = runNumber/1000000 - 1 + 2000;
+			const int date       = yearNumber*1000*10 + 101; // year + 01-01(Date)
 
-		if (!DB) {
-			LOG_WARN << "No data set found, creating new St_db_Maker... with date/time" << date << "/" << time << endm;
-			dbMk = new St_db_Maker("db", "MySQL:StarDb", "$STAR/StarDb");
+			St_db_Maker *dbMk = new St_db_Maker("db", "MySQL:StarDb", "$STAR/StarDb");
 			dbMk->SetDebug();
-			dbMk->SetDateTime(date,time); //! event or run start time, set to your liking
+			dbMk->SetDateTime(date,1);
+			cout<<"cout for test, data: "<<date<<endl;
 			dbMk->SetFlavor("ofl");
 			dbMk->Init();
 			dbMk->Make();
-		}
 
-		if (!DB) {
-			LOG_WARN << "Failed to connect to Database!" << endm;
-			return;
-		}
+			TDataSet *DB = 0;
+			DB = dbMk->GetDataBase("Calibrations/tof/tofSimResParams");
+
+			if(!DB)
+			{
+				LOG_WARN << "ERROR: could not read requested table from DB" << endm;
+				return;
+			}
 
-		St_tofSimResParams *dataset = 0;
-		dataset = (St_tofSimResParams*) DB->Find("tofSimResParams");
-
-		if (dataset) {
-			TDatime val[3];
-			dbMk->GetValidity((TTable*)dataset,val);
-			tofSimResParams_st* table = static_cast<tofSimResParams_st*>(dataset->GetTable());
-
-      mAverageTimeResTof=0;
-			for ( int i = 0; i < 120; i++ ){ //  nTrays
-				for ( int j = 0; j < 192; j++ ){
-					size_t index = i * 120 + j;
-					params[i][j] =  table[0].resolution[index];
-          mAverageTimeResTof+=params[i][j];
-					LOG_DEBUG << "tray:" << i << ", mod cell:" << j << " = " << table[0].resolution[index]  << " == " << params[i][j] << endm;
+			St_tofSimResParams *dataset = 0;
+			dataset = (St_tofSimResParams*) DB->Find("tofSimResParams");
+
+
+			if(dataset) 
+			{
+				TDatime val[2];
+				dbMk->GetValidity((TTable*)dataset,val);
+				
+				std::cout << "Dataset validity range: [ " << val[0].GetDate() << "." << val[0].GetTime() << " - "
+					<< val[1].GetDate() << "." << val[1].GetTime() << " ] " << std::endl;
+				
```

Comment:
Is this print statement really necessary? Remove it

Reply:
Will remove it